app/scripts/metamask-controller.js

```diff
@@ -2010,7 +2013,13 @@ export default class MetamaskController extends EventEmitter {
    * @returns The result of the JSON-RPC request.
    */
   handleSnapRequest(args) {
-    this._trackSnapExportUsage(args.snapId, args.handler);
+    // We're not tracking transaction insights at this point in Flask because we eagerly fetch insights in v2.
```

Comment:
We still might wanna do that to track usage of the export. We can add a separate event for viewing the insight perhaps?

Reply:
What’s the logic behind also tracking `onTransaction` export usage in `handleSnapRequest` in flask? I suppose by only tracking it in the component we’re filtering out situations where the export returns `null`.

Reply:
We're trying to track usage of the exports, if we only track when it is viewed that seems misleading?

Reply:
Yeah, okay that makes sense. I'll add a new event.